### PR TITLE
feat: transactions page

### PR DESCRIPTION
**Pull request by lcsfelix**

### CODE REVIEW

File: new-lamassu-admin/src/components/table/TableRow.js [hunk 1]
```diff
@@ -24,7 +24,7 @@ const useStyles = makeStyles({
     boxShadow: '0 0 4px 0 rgba(0, 0, 0, 0.08)'
   },
   lg: {
-    extend: info2,
+    // extend: info2,
```
Comment (RafaelTaranto) [inline new-lamassu-admin/src/components/table/TableRow.js]: Any reason why this was commented out?

File: new-lamassu-admin/src/components/LogsDownloaderPopper.js [hunk 2]
```diff
@@ -132,36 +131,26 @@ const LogsDownloaderPopover = ({ id, open, anchorEl, onClose, logsResponse, ...p
...
 
-  const formatDateFile = date => {
-    return moment(date).format('YYYY-MM-DD_HH-mm')
-  }
-
   const handleRadioButtons = (event) => {
-    setRadioButtons(toInteger(event.target.value))
+    const radio = toInteger(event.target.value)
+    setRadioButtons(radio)
+    if (radio === 0) setRange({ from: null, to: null })
```
Comment (RafaelTaranto) [inline new-lamassu-admin/src/components/LogsDownloaderPopper.js]: why not just use boolean?

File: new-lamassu-admin/src/components/expandable-table/ExpTable.js [hunk 3]
```diff
@@ -28,7 +28,7 @@ const ExpRow = ({ id, columns, details, sizes, expanded, className, expandRow, .
 
   const detailsRowClasses = {
     [classes.detailsRow]: true,
-    [classes.hideDetailsRow]: expanded
+    [classes.hideDetailsRow]: !expanded
```
Comment (RafaelTaranto) [inline new-lamassu-admin/src/components/expandable-table/ExpTable.js]: Prevent the component from being mounted instead of using a class 

`{expanded && (<Tr....)`

File: new-lamassu-admin/src/components/expandable-table/ExpTable.js [hunk 4]
```diff
@@ -28,7 +28,7 @@ const ExpRow = ({ id, columns, details, sizes, expanded, className, expandRow, .
...
   }
 
   return (
```
Comment (RafaelTaranto) [inline new-lamassu-admin/src/components/expandable-table/ExpTable.js:28]: Why are you using uuidv1 on all the components here? You assign key for the components only for the react internals, rule of thumb is using `idx`. If you generate the key every time it will be as if you're handing a new component every time.

Comment (RafaelTaranto) [reply new-lamassu-admin/src/components/expandable-table/ExpTable.js:28]: Also I don't even see uuid lib on package.json, so it wouldn't work on production.

File: new-lamassu-admin/src/components/expandable-table/ExpTable.js [hunk 5]
```diff
@@ -41,7 +41,8 @@ const ExpRow = ({ id, columns, details, sizes, expanded, className, expandRow, .
           <button onClick={() => expandRow(id)} className={classes.expandButton}>
             {expanded && <ExpandOpenIcon />}
             {!expanded && <ExpandClosedIcon />}
-          </button></Td>
+          </button>
+        </Td>
       </Tr>
       <Tr className={classnames(detailsRowClasses)}>
         <Td size={mainWidth}>
```
Comment (RafaelTaranto) [inline new-lamassu-admin/src/components/expandable-table/ExpTable.js]: rowStates logic is awfully complex here, if we need to keep only one row open why not just have a id state with the open component? Then we don't have to keep creating a new list and mapping everything on that list to false.

File: new-lamassu-admin/src/components/expandable-table/ExpTable.js [hunk 6]
```diff
@@ -28,7 +28,7 @@ const ExpRow = ({ id, columns, details, sizes, expanded, className, expandRow, .
```
Comment (RafaelTaranto) [inline new-lamassu-admin/src/components/expandable-table/ExpTable.js:27]: This component has some serious performance issues, I'll try to lay it all out here. Some comments are not on the right place because I can only review the current PR

File: new-lamassu-admin/src/pages/ServerLogs.js [hunk 7]
```diff
@@ -108,11 +109,34 @@ const Logs = () => {
...
+      return
+    }
+
+    if (range.from && range.to) {
+      const text = logs.filter((log) => moment(log.timestamp).isBetween(range.from, range.to, 'day', '[]')).map(it => JSON.stringify(it)).join('\n')
+      const blob = new window.Blob([text], {
+        type: 'text/plain;charset=utf-8'
+      })
+      FileSaver.saveAs(blob, `${formatDateFile(range.from)}_${formatDateFile(range.to)}_server`)
+    }
```
Comment (RafaelTaranto) [inline new-lamassu-admin/src/pages/ServerLogs.js]: I've seen this logic on another place, no way to move it to the base component?

File: new-lamassu-admin/src/pages/Transactions/Address.js [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+import Popover from '../../components/Popper'
+import { ReactComponent as CopyIcon } from '../../styling/icons/action/copy/copy.svg'
+import { comet } from '../../styling/variables'
+
+const Address = ({ className, type, children, ...props }) => {
+  const [anchorEl, setAnchorEl] = useState(null)
+
+  useEffect(() => {
+    setTimeout(() => setAnchorEl(null), 3000)
+  }, [anchorEl])
```
Comment (RafaelTaranto) [inline new-lamassu-admin/src/pages/Transactions/Address.js]: Why is this necessary?

File: new-lamassu-admin/src/pages/Transactions/tx.js [hunk 9]
```diff
@@ -0,0 +1,12 @@
+function toUnit (cryptoAtoms, cryptoCode) {
+  const cryptoRec = {
+    cryptoCode: 'BTC',
+    display: 'Bitcoin',
+    code: 'bitcoin',
+    unitScale: 8
+  }
```
Comment (RafaelTaranto) [inline new-lamassu-admin/src/pages/Transactions/tx.js]: Why only btc here?

**Review by RafaelTaranto — CHANGES_REQUESTED** [review]

File: new-lamassu-admin/src/components/LogsDownloaderPopper.js (hunk 2 above)
Comment (lcsfelix) [reply new-lamassu-admin/src/components/LogsDownloaderPopper.js]: The RadioGroup component is prepared for more than 2 options. Using the `value` identifier even when only two options are used is more consistent.

File: new-lamassu-admin/src/pages/Transactions/Address.js (hunk 8 above)
Comment (lcsfelix) [reply new-lamassu-admin/src/pages/Transactions/Address.js]: This auto closes the 'Copied to clipboard!' popper after 3s. The Material-Ui Popper uses a toggle anchor, so the user would have to click the Copy button again to close the popper. I think it's also a nice touch.

File: new-lamassu-admin/src/components/table/TableRow.js (hunk 1 above)
Comment (lcsfelix) [reply new-lamassu-admin/src/components/table/TableRow.js]: Hum... I'm sure there was, but now uncommenting doesn't seem to make a difference, so I'll revert it.

File: new-lamassu-admin/src/components/LogsDownloaderPopper.js (hunk 2 above)
Comment (RafaelTaranto) [reply new-lamassu-admin/src/components/LogsDownloaderPopper.js]: better to have name values, then.

File: new-lamassu-admin/src/components/table/TableRow.js (hunk 1 above)
Comment (joshmh) [reply new-lamassu-admin/src/components/table/TableRow.js]: Why not remove it, in that case?

File: new-lamassu-admin/src/components/LogsDownloaderPopper.js (hunk 2 above)
Comment (lcsfelix) [reply new-lamassu-admin/src/components/LogsDownloaderPopper.js]: Fixed in https://github.com/lamassu/lamassu-server/pull/336/commits/58c08e39d6c46210864dc97c5ae25ec7fae489b7

File: new-lamassu-admin/src/components/LogsDownloaderPopper.js [hunk 10]
```diff
@@ -121,53 +121,71 @@ const styles = {
...
-    return moment(date).format('YYYY-MM-DD_HH-mm')
+    [classes.dateRangePickerShowing]: selectedRadio === radioButtonOptions[1].value,
+    [classes.dateRangePickerHidden]: selectedRadio === radioButtonOptions[0].value
   }
 
   const handleRadioButtons = (event) => {
-    setRadioButtons(toInteger(event.target.value))
+    const radio = event.target.value
+    setSelectedRadio(radio)
   }
```
Comment (RafaelTaranto) [inline new-lamassu-admin/src/components/LogsDownloaderPopper.js]: this can be reduced to `const handleRadioButtons = compose(setSelectedRadio, get('target.value'))`

File: new-lamassu-admin/src/components/LogsDownloaderPopper.js [hunk 11]
```diff
@@ -121,53 +121,71 @@ const styles = {
 
 const useStyles = makeStyles(styles)
 
-const LogsDownloaderPopover = ({ id, open, anchorEl, onClose, logsResponse, ...props }) => {
-  const [radioButtons, setRadioButtons] = useState(0)
+const LogsDownloaderPopover = ({ id, name, open, anchorEl, getTimestamp, logs, title, ...props }) => {
+  const radioButtonOptions = [{ label: 'All logs', value: 'all' }, { label: 'Date range', value: 'range' }]
```
Comment (RafaelTaranto) [inline new-lamassu-admin/src/components/LogsDownloaderPopper.js]: Maybe create consts for `all` and `range`, that way you can reference that in code instead of `radioButtonOptions[0].value`.

File: new-lamassu-admin/src/components/expandable-table/ExpTable.js [hunk 12]
```diff
@@ -28,26 +27,29 @@ const ExpRow = ({ id, columns, details, sizes, expanded, className, expandRow, .
 
   const detailsRowClasses = {
     [classes.detailsRow]: true,
-    [classes.hideDetailsRow]: expanded
+    [classes.hideDetailsRow]: !expanded
```
Comment (RafaelTaranto) [inline new-lamassu-admin/src/components/expandable-table/ExpTable.js]: We're not using this object anymore

File: new-lamassu-admin/src/pages/Transactions/DetailsCard.js [hunk 13]
```diff
@@ -0,0 +1,209 @@
...
+              </div>
+              <div className={classes.commissionWrapper}>
+                <Label>Commission</Label>
+                <div>{`${commission} ${tx.fiatCode} (${commissionPercentage * 100} %)`}</div>
+              </div>
+            </div>
+          </div>
+          <div className={classnames(classes.col, classes.col3)}> {/* Column 3 */}
+            <div className={classnames(classes.innerRow)}>
+              <div style={{ height: 43.4 }}>
```
Comment (RafaelTaranto) [inline new-lamassu-admin/src/pages/Transactions/DetailsCard.js:152]: we should prevent using sytles, why is this div empty?

File: new-lamassu-admin/src/pages/Transactions/DetailsCard.js [hunk 14]
```diff
@@ -0,0 +1,209 @@
...
+                      ${addr.slice(5, 10)}
+                      ${addr.slice(10, 15)}
+                      ${addr.slice(15, 20)}
+                      ${addr.slice(20, 25)}
+                      ${addr.slice(25, 30)}
+                      ${addr.slice(30, 35)}
+                      ${addr.slice(35, 40)}
+                      ${addr.slice(40, 45)}
+                      ${addr.slice(45, 50)}`}
+                  </Address>
```
Comment (RafaelTaranto) [inline new-lamassu-admin/src/pages/Transactions/DetailsCard.js]: Shouldn't the address component handle the output? Also this won't work for every address, check out the `formatAddress` function on `Funding.js`

File: new-lamassu-admin/src/pages/Transactions/DetailsCard.js [hunk 15]
```diff
@@ -0,0 +1,209 @@
...
+          <div className={classnames(classes.col, classes.col2)}> {/* Column 2 */}
+            <div className={classes.innerRow}>
+              <div>
+                <Label>Transaction ID</Label>
+                <div>
+                  <Address type='txId'>
+                    {txHash &&
+                      `${txHash.slice(0, 41)}
+                      ${txHash.slice(41)}`}
+                  </Address>
```
Comment (RafaelTaranto) [inline new-lamassu-admin/src/pages/Transactions/DetailsCard.js]: Why is this using the address component? If it's for both address and txhash shouldn't it be called something else? Something like copytoclipboard or the like.

As far as formatting goes I think it's better to make the div split the hash with break lines instead of doing it by hand.

File: new-lamassu-admin/src/pages/Transactions/DetailsCard.js [hunk 16]
```diff
@@ -0,0 +1,209 @@
...
+          </div>
+          <div className={classnames(classes.col, classes.col3)}> {/* Column 3 */}
+            <div className={classes.innerRow}>
+              <div>
+                <Label>Session ID</Label>
+                <Address type='sessionId'>
+                  {tx.id &&
+                    `${tx.id.slice(0, 21)}
+                  ${tx.id.slice(21)}`}
+                </Address>
```
Comment (RafaelTaranto) [inline new-lamassu-admin/src/pages/Transactions/DetailsCard.js]: Same as above

File: new-lamassu-admin/src/pages/Transactions/Transactions.js [hunk 17]
```diff
@@ -0,0 +1,206 @@
...
+      textAlign: 'center'
+    },
+    {
+      value: ''
+    }
+  ]
+
+  const rows = txResponse && txResponse.data.map(tx => {
+    const customerName = tx.customerName ? tx.customerName
+      : (tx.customerIdCardData ? `${startCase(lowerCase(tx.customerIdCardData.firstName.slice(0, 1)))}. ${startCase(lowerCase(tx.customerIdCardData.lastName))}` : tx.customerPhone)
```
Comment (RafaelTaranto) [inline new-lamassu-admin/src/pages/Transactions/Transactions.js]: This is complicated to read, split into a couple of functions.

File: new-lamassu-admin/src/pages/Transactions/Transactions.js [hunk 18]
```diff
@@ -0,0 +1,206 @@
...
+    180, // Machine
+    162, // Customer
+    110, // Cash
+    141, // Crypto
+    136, // Address
+    124, // Date
+    124, // Time
+    90, // Trade
+    71 // Expand
+  ]
```
Comment (RafaelTaranto) [inline new-lamassu-admin/src/pages/Transactions/Transactions.js]: I think it makes more sense to pass down the sizes with the main col object

```
 {
          value: moment(tx.created).format('HH:mm:ss'),
          textAlign: 'right',
          size: 12398612
        },
```

File: new-lamassu-admin/src/pages/Transactions/Transactions.js [hunk 19]
```diff
@@ -0,0 +1,206 @@
...
+    }
+
+    if (range.from && range.to) {
+      const text = txs.filter((tx) => moment(tx.created).isBetween(range.from, range.to, 'day', '[]')).map(it => JSON.stringify(it)).join('\n')
+      const blob = new window.Blob([text], {
+        type: 'text/plain;charset=utf-8'
+      })
+      FileSaver.saveAs(blob, `${formatDateFile(range.from)}_${formatDateFile(range.to)}_transactions`)
+    }
+  }
```
Comment (RafaelTaranto) [inline new-lamassu-admin/src/pages/Transactions/Transactions.js]: Didn't we abstracted this away to a component?

**Review by RafaelTaranto — CHANGES_REQUESTED** [review]

File: new-lamassu-admin/src/pages/Transactions/CopyToClipboard.js [hunk 20]
```diff
@@ -0,0 +1,89 @@
...
+    return address.replace(/(.{5})/g, '$1 ')
+  }
+
+  const formatTxId = (id = '') => {
+    return id.replace(/(.{41})/g, '$1 ')
+  }
+
+  const formatSessionId = (id = '') => {
+    return id.replace(/(.{21})/g, '$1 ')
+  }
```
Comment (RafaelTaranto) [inline new-lamassu-admin/src/pages/Transactions/CopyToClipboard.js]: Still not what I meant. This component should do what it says it does, Copy to Clipboard. Formatting should be handled externally. Format address is something that we need, but the other two are just line breaking at the end of the div size, so you should make it break automatically instead of using a function for this.

File: new-lamassu-admin/src/pages/Transactions/Transactions.js [hunk 21]
```diff
@@ -0,0 +1,159 @@
...
+  const { response: txResponse } = useAxios({
+    url: 'http://localhost:8070/api/txs/',
+    method: 'GET',
+    trigger: []
+  })
+
+  const formatCustomerName = (customer) => {
+    const { firstName, lastName } = customer
+
+    return `${startCase(lowerCase(firstName.slice(0, 1)))}. ${startCase(lowerCase(lastName))}`
```
Comment (RafaelTaranto) [inline new-lamassu-admin/src/pages/Transactions/Transactions.js]: Why not do this:

```startCase(lowerCase(`${firstName.slice(0, 1)}. ${lastName}`))```

File: new-lamassu-admin/src/pages/Transactions/Transactions.js [hunk 22]
```diff
@@ -0,0 +1,159 @@
...
+        },
+        {
+          name: 'Crypto',
+          value: `${toUnit(new BigNumber(tx.cryptoAtoms), tx.cryptoCode).toFormat(5)} ${tx.cryptoCode}`,
+          textAlign: 'right',
+          size: 141
+        },
+        {
+          name: 'Address',
+          value: `${tx.toAddress.slice(0, 9)}...`,
```
Comment (RafaelTaranto) [inline new-lamassu-admin/src/pages/Transactions/Transactions.js]: this can be done with css as well, it's usually better since we don't have to change here if we ever need to change this size.

File: new-lamassu-admin/src/pages/Transactions/Transactions.js [hunk 23]
```diff
@@ -0,0 +1,159 @@
...
+          size: 136
+        },
+        {
+          name: 'Date (UTC)',
+          value: moment(tx.created).format('YYYY-MM-D'),
```
Comment (RafaelTaranto) [inline new-lamassu-admin/src/pages/Transactions/Transactions.js]: does moment set UTC correctly?

**Review by RafaelTaranto — CHANGES_REQUESTED** [review]

File: new-lamassu-admin/src/pages/Transactions/Transactions.js (hunk 21 above)
Comment (lcsfelix) [reply new-lamassu-admin/src/pages/Transactions/Transactions.js]: `startCase` removes the dot after the first name.

File: new-lamassu-admin/src/pages/Transactions/Transactions.js (hunk 23 above)
Comment (lcsfelix) [reply new-lamassu-admin/src/pages/Transactions/Transactions.js]: I assumed the date was already stored in the DB in the proper format, but I can use `moment.utc` to make sure.

**Comment by lcsfelix** [comment]
[Create transactions page](https://trello.com/c/iEaZLvTE/119-create-transactions-page)

**Comment by lcsfelix** [comment]
[Create transactions page](https://trello.com/c/iEaZLvTE/119-create-transactions-page)

File: new-lamassu-admin/src/pages/Transactions/Transactions.js (hunk 21 above)
Comment (lcsfelix) [reply new-lamassu-admin/src/pages/Transactions/Transactions.js]: https://github.com/lamassu/lamassu-server/pull/336/commits/7ecd04765289b4680c8ff903ecee0d0da44af8e6 removes formatting the last name altogether, to prevent misformatting names with special characters, such as [Double-barreled names](https://en.wikipedia.org/wiki/Double-barrelled_name).

**Review by RafaelTaranto — APPROVED** [review]
lgtm

**Comment by RafaelTaranto** [comment]
Oh wait, this should be opened to dev.
